### PR TITLE
Post FluxC vendor improvements: remove unused Wiremock files, migrate to local version catalog

### PR DESCRIPTION
**Pull request by wzieba**
### Description

This PR adds several, post FluxC vendor improvements:
- remove `*/wc/*` WireMock files that are simply not needed (I should've exclude them during migration but I haven't noticed them)
- clean `gradle.properties-example` of vendored modules
- migrate `fluxc` modules to use local version catalog instead of shared dependency catalog
As with the FluxC merge, this doesn't seem to add many benefits and makes updating dependencies and keeping consise dependency tree more difficult.
 
### Testing

This is build-focus work. Green light from CI is completely fine. Mind the dependency diff tree comment: this PR shouldn't introduce changes to the runtime classpath.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: libs/fluxc/build.gradle [hunk 1]
```diff
@@ -71,70 +71,68 @@ android.buildTypes.all { buildType ->
...
+    implementation libs.apache.commons.text
+    api libs.androidx.paging.runtime
+    implementation libs.androidx.room.runtime
+    kapt libs.androidx.room.compiler
+    implementation libs.androidx.room.ktx
 
     // Dagger
-    implementation sharedLibs.google.dagger
-    kapt sharedLibs.google.dagger.compiler
-    compileOnly sharedLibs.glassfish.javax.annotation
```
Comment (ParaskP7) [inline libs/fluxc/build.gradle:108]: Praise (❤️): Good call! 💯

File: libs/fluxc/build.gradle [hunk 2]
```diff
@@ -71,70 +71,68 @@ android.buildTypes.all { buildType ->
...
 
     // Unit tests
-    testImplementation sharedLibs.junit
-    testImplementation sharedLibs.kotlin.test.junit
-    testImplementation sharedLibs.kotlinx.coroutines.test
-    testImplementation sharedLibs.androidx.test.core
-    testImplementation sharedLibs.robolectric
-    testImplementation sharedLibs.mockito.core
-    testImplementation sharedLibs.mockito.kotlin
-    testImplementation sharedLibs.mockito.inline
```
Comment (ParaskP7) [inline libs/fluxc/build.gradle:126]: Praise (❤️): Good call! 💯

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -14,23 +14,28 @@ androidx-cardview = '1.0.0'
 androidx-compose-bom = '2024.08.00'
```
Comment (ParaskP7) [inline gradle/libs.versions.toml:14]: Praise (❤️): Thanks for sorting everything alphabetically at the very end, fixing my previous mistakes! 🙇

**Review by ParaskP7 — APPROVED** [review]
👋 @wzieba !

I have reviewed and quickly smoke tested this PR too, everything works as expected, great job on removing the need for [ADC](https://github.com/Automattic/android-dependency-catalog) on this project! 🌟 

> Mind the dependency diff tree comment: this PR shouldn't introduce changes to the runtime classpath.

👍 on [that](https://github.com/wordpress-mobile/WordPress-Android/pull/21353#issuecomment-2435711726), but also, me seeing `dagger` pointing to `2.52`, it made me wonder why... Only to then understand that this is coming from the [login](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/libs/login/build.gradle#L49) module. Having the `dagger` version being driven by this module instead of version catalogs seems a bit odd, right.  On a separate PR, do you maybe have any plans to move these version into version catalogs as well? 🤔

File: gradle/libs.versions.toml (hunk 3 above)
Comment (wzieba) [reply gradle/libs.versions.toml:14]: FYI I use String Manipulation IDEA plugin for things like this!

Comment (ParaskP7) [reply gradle/libs.versions.toml:14]: TIL about [this](https://plugins.jetbrains.com/plugin/2162-string-manipulation) plugin, I wasn't aware of this, just tried it and it works like a charm, thanks for sharing! 🙇 ❤️

**Comment by ParaskP7** [comment]
> 👍 on https://github.com/wordpress-mobile/WordPress-Android/pull/21353#issuecomment-2435711726, but also, me seeing dagger pointing to 2.52, it made me wonder why... Only to then understand that this is coming from the [login](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/libs/login/build.gradle#L49) module. Having the dagger version being driven by this module instead of version catalogs seems a bit odd, right. On a separate PR, do you maybe have any plans to move these version into version catalogs as well? 🤔

FYI: This work is actually being address here: #21367 ✅

**Pull request merged**